window_manager/generate-schema-validator.mjs

```diff
@@ -0,0 +1,155 @@
+import {writeFileSync} from 'fs';
+import Ajv from 'ajv';
+import standaloneCode from 'ajv/dist/standalone/index.js';
+
+/**
+ * Run with
+ *
+ * node generate-schema-validator.mjs
+ *
+ * to build the jsoneditor/schema-validator.js file.
+ */
+
+
+/**
+ * Definitions of schema objects.
+ *
+ * @see https://ajv.js.org/json-schema.html#json-data-type
+ */
+const SCHEMA_DEFINITIONS = [
+  {
+    $id: 'startend',
+    title: 'Specification of pixel position in String (%) or number of pixels',
+    oneOf: [
+      {
+        type: 'number',
+        minimum: 0,
+      },
+      {
+        type: 'string',
+        pattern: '^\\d+(\\.\\d+)?%$',
+        minLength: 1,
+      },
+    ],
+  },
+  {
+    $id: 'rowcol',
+    title: 'Specification of start/end positions for row/column values',
+    type: 'object',
+    properties: {
+      comment: {type: 'string'},
+      start: {$ref: 'startend'},
+      end: {$ref: 'startend'},
+    },
+    additionalProperties: false,
+    required: ['start', 'end'],
+  },
+  {
+    $id: 'action',
+    title: 'Definition of the action object type',
+    type: 'object',
+    properties: {
+      comment: {type: 'string'},
+      id: {type: 'string', minLength: 1},
+      display: {type: 'string', minLength: 1},
+      menuName: {type: 'string', minLength: 1},
+      shortcutId: {type: 'integer', minimum: 1, maximum: 9},
+      column: {$ref: 'rowcol'},
+      row: {$ref: 'rowcol'},
+    },
+    additionalProperties: false,
+    required: ['id', 'display', 'row', 'column'],
+  }, {
+    $id: 'matcher',
+    title: 'Definition of the matcher object type',
+    type: 'object',
+    properties: {
+      comment: {type: 'string'},
+      actions: {
+        type: 'array',
+        items: {type: 'string', minLength: 1},
+        minItems: 1,
+        uniqueItems: true,
+      },
+      windowTypes: {
+        type: 'array',
+        items: {
+          type: 'string',
+          enum: ['normal', 'popup', 'panel', 'app', 'devtools'],
+        },
+      },
+      anyTabUrl: {type: 'string', minLength: 1},
+      minTabsNum: {type: 'integer', minimum: 0},
+      maxTabsNum: {type: 'integer', minimum: 0},
+    },
+    additionalProperties: false,
+    required: ['actions'],
+  },
+  {
+    $id: 'actions',
+    type: 'array',
+    items: {$ref: 'action'},
+  },
+  {
+    $id: 'matchers',
+    type: 'array',
+    items: {$ref: 'matcher'},
+  },
+  {
+    $id: 'settings',
+    type: 'object',
+    title: 'Definition of the settings object type',
+    properties: {
+      popupButtonColor: {type: 'string', minLength: 1},
+      popupBackgroundColor: {type: 'string', minLength: 1},
```

Comment:
yesterday I added `rememberPositionsSetWithShortcut` action - please add it here too